tests/test_runs/test_run_functions.py

```diff
@@ -342,7 +341,9 @@ def test_run_regression_on_classif_task(self):
 
         clf = LinearRegression()
         task = openml.tasks.get_task(task_id)
-        with self.assertRaises(AttributeError):
+        # internally dataframe is loaded and targets are categorical
+        # which LinearRegression() cannot handle
+        with self.assertRaises(ValueError):
```

Comment:
Is it possible that you use a different task which has no categories/categoricals to keep the original behavior? Could you then please also make this `self.assertRaisesRegex` so that we know the error that's happening?

Reply:
Just changed the assert to `assertRaisesRegex`.
From what I am gathering, this test is designed to fail when running `LinearRegression` on a classification task since the target variables for a classification task should be categorical. I am not sure how we can take a classification task without categorical targets. Or do you mean, finding a dataset where the categories are numeric and represented so as well? In that case, do we expect or want the test to pass?

Reply:
The original error was that it was an `AttributeError`, and it would be great if we keep that (I just don't remember what attribute failed). By using a different task I was hoping that the target would be encoded as a numpy array with integers/floats. Is this somehow possible?

Reply:
@mfeurer 
enforced a numpy array load of the dataset in the task run, and we get the `AttributeError`. So I updated the test case to catch an `AttributeError` and it passes.
Hope this suffices for now for this PR.

The rest of the failing test cases are I think dependent on other issues that need fixing. The changes this PR intends to implement, in my opinion, looks fine for now.